features/frameworks/greentea-client/source/greentea_metrics.cpp

```diff
@@ -20,6 +20,8 @@
 #include "platform/mbed_stats.h"
 #include <stdint.h>
 
+#if defined(MBED_CONF_RTOS_PRESENT) 
```

Comment:
This `#if` define would be a bit overlap on the logical  with later `line 35`, I think you can combine those `#if`  together

Reply:
I think you can get CPU stats in baremetal if the device has DEVICE_SLEEP and DEVICE_LPTICKER.

Reply:
Moved the RTOS supported metrics under MBED_CONFIG_RTOS_PRESENT macro to indirectly disable for bare metal.